File: Session-3/DDD-Sample-codes/rtgs-return-management-dotnet10/Infrastructure/Persistence/InMemoryRtgsReturnRepository.cs
using System.Collections.Concurrent;
using Rtgs.ReturnManagement.Api.Domain.Model;
using Rtgs.ReturnManagement.Api.Domain.Ports;

namespace Rtgs.ReturnManagement.Api.Infrastructure.Persistence;

public sealed class InMemoryRtgsReturnRepository : IRtgsReturnRepository
{
    private readonly ConcurrentDictionary<Guid, RtgsReturn> _store = new();

    public RtgsReturn Save(RtgsReturn rtgsReturn)
    {
        _store[rtgsReturn.Id] = rtgsReturn;
        return rtgsReturn;
    }

    public RtgsReturn? GetById(Guid id) => _store.TryGetValue(id, out var rtgsReturn) ? rtgsReturn : null;
}

File: Session-3/DDD-Sample-codes/rtgs-return-management-springboot/src/main/java/com/bank/rtgs/returnmanagement/infrastructure/persistence/InMemoryRtgsReturnRepository.java
package com.bank.rtgs.returnmanagement.infrastructure.persistence;

import com.bank.rtgs.returnmanagement.domain.model.RtgsReturn;
import com.bank.rtgs.returnmanagement.domain.ports.RtgsReturnRepository;
import org.springframework.stereotype.Repository;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

@Repository
public class InMemoryRtgsReturnRepository implements RtgsReturnRepository {
    private final Map<UUID, RtgsReturn> store = new ConcurrentHashMap<>();

    @Override
    public RtgsReturn save(RtgsReturn rtgsReturn) {
        store.put(rtgsReturn.getId(), rtgsReturn);
        return rtgsReturn;
    }

    @Override
    public Optional<RtgsReturn> findById(UUID id) {
        return Optional.ofNullable(store.get(id));
    }
}
